src/averageAges.js

```diff
@@ -7,19 +7,33 @@
  * function calculates average age only for men who died in this century
  *
  * To calculate century:
- * Divide year of person's death by 100: Math.ceil(person.died / 100)
+  * Divide year of person's death by 100: Math.ceil(person.died / 100)
+  *
+  // learn how to use array methods like .filter .map .some .every .find .reduce
+  // avoid using loop and forEach
+  // replace `if ()` statement with &&, || or ?:
+  // without nesting
  *
  * @param {object[]} people
  * @param {number} century - optional
  *
  * @return {number}
  */
+const averageAge = (people) => {
+  if (people.length === 0) {
+    return 0;
+  }
+
+  return people.reduce((sum, person) => sum + person.died - person.born, 0)
+  / people.length;
+};
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(person => person.sex === 'm');
+  const menDiedInCentury = men
+    .filter(person => Math.ceil(person.died / 100) === century);
```

Comment:
now function does both of these filters. Better to check which filter to apply and only then to apply it (at all functions)